### PR TITLE
Expose CommandPermission in NoPermissionException

### PR DESCRIPTION
**Pull request by Konicai**
Not quite sure how I feel about the new method being fluent, while the other exceptions don't use fluent getters. This seemed like the best name though given what there is to work with.

### CODE REVIEW

**Comment by Konicai** [comment]
i believe i'll need to update the exception handlers of each platform to not use the deprecated method - i'll do so asap if this change is accepted

File: cloud-core/src/main/java/cloud/commandframework/exceptions/NoPermissionException.java [hunk 1]
```diff
@@ -67,11 +67,25 @@ public final String getMessage() {
...
         return this.missingPermission.toString();
     }
 
+    /**
+     * Returns the missing {@link CommandPermission}
+     *
+     * @return the missing permission
+     * @since 1.8.5
+     */
+    @API(status = API.Status.STABLE, since = "1.8.5")
```
Comment (jpenilla) [inline cloud-core/src/main/java/cloud/commandframework/exceptions/NoPermissionException.java:84]: new API should target 1.9

Comment (Konicai) [reply cloud-core/src/main/java/cloud/commandframework/exceptions/NoPermissionException.java:84]: okay cool - `1.9.0` ?

Comment (Konicai) [reply cloud-core/src/main/java/cloud/commandframework/exceptions/NoPermissionException.java:84]: and the version for the deprecation annotation should also be `1.9.0` ?

Comment (jpenilla) [reply cloud-core/src/main/java/cloud/commandframework/exceptions/NoPermissionException.java:84]: yes and yes

File: cloud-core/src/main/java/cloud/commandframework/exceptions/NoPermissionException.java [hunk 2]
```diff
@@ -67,11 +67,25 @@ public final String getMessage() {
...
+    @Deprecated
     public @NonNull String getMissingPermission() {
         return this.missingPermission.toString();
     }
 
+    /**
+     * Returns the missing {@link CommandPermission}
+     *
+     * @return the missing permission
+     * @since 1.8.5
```
Comment (jpenilla) [inline cloud-core/src/main/java/cloud/commandframework/exceptions/NoPermissionException.java:82]: missed a spot

**Review by jpenilla — APPROVED** [review]

**Pull request merged**